Parser/src/main/java/Parser.java

```diff
@@ -0,0 +1,297 @@
+        this.generateCanonicalCollection();
+        this.table = new ParsingTable(grammar.getAllSymbols(), this.states.size());
+        for (var index = 0; index < states.size(); index++) {
+            var state = states.get(index);
+            var actions = new HashMap<Item, String>();
+            for (var item : state) {
+                if (item.getInput().isEmpty()) {
+                    if (item.getLeft().equals(grammar.getInitialSymbol())) {
+                        actions.put(item, "accept");
+                    } else {
+                        for (int i = 0; i < grammar.getProductions().size(); i++) {
+                            var prod = grammar.getProductions().get(i);
+                            if (prod.getLeftForCFG().equals(item.getLeft()) && prod.right.equals(item.getWork())) {
+                                actions.put(item, "reduce" + (i + 1));
+                                break;
+                            }
+                        }
+                    }
+                } else {
+                    actions.put(item, "shift");
+                }
+            }
+            var action = actions.entrySet().iterator().next().getValue();
+            for (var a : actions.keySet()) {
+                if (!actions.get(a).equals(action)) {
+                    throw new ParsingTableConflictException(state + "has action " + actions.get(a) + "on item" + a.toString() + " conflicts " + action + "\n");
+                }
+            }
+            table.getRows().get(index).setAction(action);
+
+            for (var symbol : grammar.getAllSymbols()) {
+                var goToResult = goTo(state, symbol);
+                var goToState = this.findState(goToResult);
+                if (goToState != -1) {
+                    table.setGoToSymbol(index, symbol, goToState);
+                }
+            }
+        }
+    }
+
+    private void actionShift() {
+        var currentState = workingStack.peek();
+        var currentSymbol = inputStack.peek();
+        var goToState = table.getRows().get(currentState).getGoToState(currentSymbol);
+        inputStack.pop();
+        workingStack.push(currentSymbol);
+        workingStack.push(goToState);
+    }
+
+    private void actionReduce() {
+        var currentState = workingStack.peek();
+        var productionPos = Integer.parseInt(table.getRows().get(currentState).getAction().substring(6));
+        var production = grammar.getProductions().get(productionPos - 1);
+        var symbols = new ArrayList<>(production.getRight());
+        var symbolsStack = new Stack<Symbol>();
+        symbolsStack.addAll(symbols);
+        while (!symbolsStack.isEmpty()) {
+            try {
+                var currentS = workingStack.pop();
+                if (currentS.equals(symbolsStack.peek().getLabel())) {
+                    symbolsStack.pop();
+                }
+            } catch (Exception e) {
+                System.out.println("e");
```

Comment:
you probably want to print the exception here
you should also never do catch (Exception ...)